hail/src/main/scala/is/hail/expr/ir/agg/NDArraySumAggregator.scala

```diff
@@ -0,0 +1,125 @@
+package is.hail.expr.ir.agg
+
+import is.hail.annotations.{Region, StagedRegionValueBuilder}
+import is.hail.asm4s._
+import is.hail.expr.ir.functions.StringFunctions
+import is.hail.expr.ir.{EmitCode, EmitCodeBuilder, EmitRegion, coerce}
+import is.hail.types.physical.{PBooleanRequired, PCanonicalTuple, PCode, PNDArray, PNDArrayCode, PNDArrayValue, PNumeric, PType}
+import is.hail.utils._
+
+class NDArraySumAggregator (ndTyp: PNDArray) extends StagedAggregator {
+  override type State = TypedRegionBackedAggState
+
+  // State needs:
+  // 1. Initialized or not
+  // 2. The ndarray itself.
+
+  override def resultType: PType = ndTyp
+
+  val stateType = PCanonicalTuple(true, PBooleanRequired, ndTyp)
```

Comment:
Because the seqOp skips missing values, the ndarray field can never be missing if the state is initialized, right? Is there a reason not to just let a missing ndarray be the uninitialized state? Then the result is exactly the state.

If not, up to you if you want to change it. I think it's a bit cleaner, but not a big difference.